RecoMuon/MuonRechitClusterProducer/plugins/CSCRechitClusterProducer.cc

```diff
@@ -0,0 +1,237 @@
+        int stationRing = endcap * (CSCDetId::station(cscdetid) * 10 + CSCDetId::ring(cscdetid));
+        if (CSCDetId::ring(cscdetid) == 4)
+          stationRing = endcap * (CSCDetId::station(cscdetid) * 10 + 1);  // ME1/a has ring==4
+        station_count_map[CSCDetId::station(cscdetid)]++;
+        if (abs(stationRing) == 11)
+          nME11++;
+        if (abs(stationRing) == 12)
+          nME12++;
+        time += (rechit->tpeak() + rechit->wireTime());
+        time_strip += rechit->tpeak();
+      }
+      //station statistics
+      std::map<int, int>::iterator it;
+      for (it = station_count_map.begin(); it != station_count_map.end(); it++) {
+        if (it->second >= 10) {
+          nStation10++;
+          avgStation10 += (it->first) * (it->second);
+          totStation10 += (it->second);
+        }
+      }
+      if (nStation10 != 0) {
+        avgStation10 = avgStation10 / totStation10;
+      }
+      time = time / (2 * rechits.size());
+      time_strip = time_strip / (rechits.size());
+
+      //derive cluster statistics
+      for (auto& rechit : rechits) {
+        timeSpread += (rechit->tpeak() - time_strip) * (rechit->tpeak() - time_strip);
+      }
+      timeSpread = std::sqrt(timeSpread / rechits.size());
+
+      float jetX = fjJet.px() / rechits.size();
+      float jetY = fjJet.py() / rechits.size();
+      float jetZ = fjJet.pz() / rechits.size();
+
+      reco::MuonCSCRecHitCluster cls(
+          jetX, jetY, jetZ, rechits.size(), nStation10, avgStation10, time, timeSpread, nME11, nME12, rechits);
+      //std::cout<<"CSCrechitCluster" ;
+      //std::cout<< " my phi =  " <<   phi  << " my eta =  " <<    eta ;
+      //std::cout<< " cls phi =  " <<    cls.phi()  << " cls eta =  " <<    (cls.eta())
+      //std::cout<< " cls nStation10 =  " <<    cls.nStation10()  << " cls avgStation10 =  " <<    (cls.avgStation10());
+      //std::cout<< " cls time =  " <<    cls.time()  << " cls timeSpread =  " <<    (cls.timeSpread()) <<std::endl;
+
+      CSCclusters->push_back(cls);
+    }
+  }
+  ev.put(std::move(CSCclusters));
+}
+
+std::vector<rechitRef> CSCRechitClusterProducer::getConstituents(const std::vector<fastjet::PseudoJet>& fjConstituents) {
+  std::vector<rechitRef> result;
+  for (unsigned int i = 0; i < fjConstituents.size(); i++) {
+    auto index = fjConstituents[i].user_index();
+    if (index >= 0 && static_cast<unsigned int>(index) < inputs_.size()) {
+      result.emplace_back(inputs_[index]);
+    }
+  }
+  return result;
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void CSCRechitClusterProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<int>("nRechitMin", 50);
+  desc.add<double>("rParam", 0.4);
+  desc.add<edm::InputTag>("cscRecHitLabel", edm::InputTag("csc2DRecHits"));
+  descriptions.add("CSCRechitClusterProducer", desc);
```

Comment:
```suggestion
  descriptions.add("cscRechitClusterProducer", desc);
```
to follow the lower case rule for instance names